### PR TITLE
Delete data folder

### PR DESCRIPTION
**Pull request by nwlandry**
The disGene dataset is now in XGI-DATA (https://gitlab.com/complexgroupinteractions/xgi-data-disgenenet) and all references have been removed to it in the library.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -67,6 +67,8 @@ mind.
 - PyPI: https://pypi.org/project/xgi/
 - Documentation: https://xgi.readthedocs.io/
 
+Sign up for our `mailing list <http://eepurl.com/igE6ez>`_!
```
Comment (maximelucas) [inline docs/source/index.rst:70]: noice

**Review by maximelucas — APPROVED** [review]

File: docs/source/index.rst (hunk 1 above)
Comment (maximelucas) [reply docs/source/index.rst:70]: Should we add a link to twitter/mastodon somewhere too?

Comment (nwlandry) [reply docs/source/index.rst:70]: Sure!

Comment (nwlandry) [reply docs/source/index.rst]: Done.

**Pull request merged**